TelCo.ColorCoder/Pair.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Diagnostics;
+using System.Drawing;
+
+namespace TelCo.ColorCoder
+{  
+	class Pair
+	{
+		public Color majorColor;
+        	public Color minorColor;
+		public override string ToString()
+		{
+		    return string.Format("MajorColor:{0}, MinorColor:{1}", majorColor.Name, minorColor.Name);
+		}
+        public static int GetPairNumberFromColor(Pair pair)
+        {
+            // Find the major color in the array and get the index
+            int majorIndex = -1;
+            for (int i = 0; i < getColor.colorMapMajor.Length; i++)
+            {
+                if (getColor.colorMapMajor[i] == pair.majorColor)
+                {
+                    majorIndex = i;
+                    break;
+                }
+            }
+            // Find the minor color in the array and get the index
+            int minorIndex = -1;
+            for (int i = 0; i < getColor.colorMapMinor.Length; i++)

```

Comment:
This `for` loop is almost identical to the previous one. Only difference is the array being searched.
Notice the comments above the loops. Try turning them into functions.

Reply:
Correct. Working on it